CHANGELOG.md

```diff
@@ -33,10 +33,12 @@
 * `New()` will no longer return a broken `*Client` in some instances.
 * Incoming transfer frames received during initial link detach are no longer discarded.
 * Session will no longer flood peer with flow frames when half its incoming window is consumed.
+* Newly created `Session` won't leak if the context passed to `Client.NewSession()` expires before exit.
 
 ### Other Changes
 * Errors when reading/writing to the underlying `net.Conn` are now wrapped in a `ConnectionError` type.
 * Disambiguate error message for distinct cases where a session wasn't found for the specified remote channel.
 * Removed `link.Paused` as it didn't add much value and was broken in some cases.
 * Only send one flow frame when a drain has been requested.
 * Session window size increased to 5000.
+* Creation and deletion of `Session` instances have been made deterministic.
```

Comment:
I'm going to look at moving this out of this PR so that it can go directly into `main`.

Reply:
Done, see https://github.com/Azure/go-amqp/pull/177